internal/commands/function/command.go

```diff
@@ -0,0 +1,134 @@
+package function
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/briandowns/spinner"
+
+	"github.com/spf13/pflag"
+)
+
+// set of supported `push` command strings
+const (
+	CommandUse = "function run"
+)
+
+// set of supported `push` command strings
+var (
+	CommandAliases = []string{"run"}
+)
+
+// Command is the `push` command
+type Command struct {
+	inputs inputs
+}
+
+// Flags is the command flags
+func (cmd *Command) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+
+	fs.StringVar(&cmd.inputs.Name, flagFunctionName, "", flagFunctionNameUsage)
+	fs.StringArrayVar(&cmd.inputs.Args, flagFunctionArgs, nil, flagFunctionArgsUsage)
+	fs.StringVar(&cmd.inputs.User, flagAsUser, "", flagAsUserUsage)
+}
+
+// Inputs is the command inputs
+func (cmd *Command) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+func isJSON(data string) bool {
+	var obj map[string]interface{}
+	if err := json.Unmarshal([]byte(data), &obj); err == nil {
+		return true
+	}
+	var list []interface{}
+	if err := json.Unmarshal([]byte(data), &list); err == nil {
+		return true
+	}
+	return false
+}
+
+func isInt(data string) bool {
+	if _, err := strconv.Atoi(data); err != nil {
+		return false
+	}
+	return true
+}
+
+func isFloat(data string) bool {
+	if _, err := strconv.ParseFloat(data, 64); err != nil {
+		return false
+	}
+	return true
+}
+
+// Handler is the command handler
+func (cmd *Command) Handler(profile *cli.Profile, ui terminal.UI, clients cli.Clients) error {
+	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	function, err := cmd.inputs.ResolveFunction(ui, clients.Realm, app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	args := make([]interface{}, 0, len(cmd.inputs.Args))
+	if cmd.inputs.Args != nil {
+		for _, arg := range cmd.inputs.Args {
```

Comment:
why not just pass this straight to the request instead of this additional processing?

Reply:
The the passed values would only ever be a single string per arg. 
For example a function that returns the args passed in (there are 2 args):
`  --args='{"value1": 1,"abcs": ["x", "y", "z"]}' --args='[1,2]'`
Results of straight passing:
```
14:02:25 UTC INFO  Result
{
  "arg1": "{\"value1\": 1,\"abcs\": [\"x\", \"y\", \"z\"]}",
  "arg2": "[1,2]"
}
```
Results with parsed conversion logic:
```
14:04:48 UTC INFO  Result
{
  "arg1": {
    "abcs": [
      "x",
      "y",
      "z"
    ],
    "value1": {
      "$numberInt": "1"
    }
  },
  "arg2": [
    {
      "$numberInt": "1"
    },
    {
      "$numberInt": "2"
    }
  ]
}
```